recipes/gtsam/all/conanfile.py

```diff
@@ -121,6 +121,10 @@ def validate(self):
         if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
             raise ConanInvalidConfiguration ("GTSAM requires MSVC >= 15")
 
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.version) >= '4.1' \
```

Comment:
```suggestion
        if is_msvc(self) and tools.Version(self.version) >= '4.1' \
```
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc

Don't forget to add the new import

Reply:
Very nice! Pushed a commit which replaces the manual checks with `is_msvc()` now.